### ISSUE TITLE
Incorrect exactPattern regex for Maestro

### ISSUE THREAD

**Issue by snakehill** [braintree/credit-card-type#51]
Hereby the current patterns from within index.js:

```
 prefixPattern: /^(5|5[06-9]|6\d*)$/
 exactPattern: /^5[06-9]\d*$/
```

Unlike the prefixPattern, the exactPattern does not support any number starting with a 6, including but not limited to most of the Maestro test numbers out there, like the following:

- 6304000000000000
- 6759649826438453
- 6799990100000000019
- 6763306505322185
- 6762026435590492

There is a conflict with Discover (and its partnerships, like that with UnionPay, see https://www.discovernetwork.com/downloads/IPP_VAR_Compliance.pdf) in the 60XXXX to 64XXXX range, but I doubt Maestro's 6 IIN range was supposed to be neglected altogether.

**Comment by crookedneighbor** [comment]
I've opened a PR to clear up some of this: https://github.com/braintree/credit-card-type/pull/52

A bit of an explanation on how `exactPattern` and `prefixPattern` are [used internally](https://github.com/braintree/credit-card-type/pull/52). 

```js
if (value.exactPattern.test(cardNumber)) {
      exactResults.push(clone(value));
    } else if (value.prefixPattern.test(cardNumber)) {
      prefixResults.push(clone(value));
    }
  }

  return exactResults.length ? exactResults : prefixResults;
```

The confusion is coming from the naming of the properties. `exactPattern` seems to indicate that it's a strict regex of the possible values a card can be, but that is not the case.

We can't include `^6\d*` as part of the exact pattern because that would prevent UnionPay cards from being part of the card arrays when `62` is submitted (as you noted). However, internally, that doesn't matter as the prefix pattern still catches it and supplies UnionPay and Maestro as possible card types. When we get to `62183` (no longer a potentially valid UnionPay card), the `prefixPattern` regex still catches it as a Maestro card and returns that array.

Ultimately, the pattern properties shouldn't be used. In my PR, I've made a note of that and left a todo comment to remove them from the result object in the next major version bump.

In my PR, I have updated the exact pattern for Maestro cards to include `6[37]`, as no other cards match that pattern in their `prefixPattern`, but it shouldn't actually matter for returning the Maestro card type.
